sensor/kubernetes/listener/resources/secrets.go

```diff
@@ -153,11 +154,17 @@ func DockerConfigToImageIntegration(secret *v1.Secret, registry string, dce conf
 		registryType = rhel.RedHatRegistryType
 	}
 
-	id, err := deriveIDFromSecret(secret, registry)
-	if err != nil {
-		return nil, errors.Wrapf(err, "deriving image integration ID from secret %q", secret.UID)
+	var id string
+	if features.SourcedAutogeneratedIntegrations.Enabled() {
+		id = uuid.NewV4().String()
+	} else {
+		var err error
+		id, err = deriveIDFromSecret(secret, registry)
+		if err != nil {
+			return nil, errors.Wrapf(err, "deriving image integration ID from secret %q", secret.UID)
+		}
 	}
```

Comment:
This should be flipped, right?
`id = uuid.NewV4().String()` was done before the scoped autogenerated integrations.